src/overlay/Peer.cpp

```diff
@@ -799,6 +818,20 @@ Peer::recvSCPMessage(StellarMessage const& msg)
     if (res != Herder::ENVELOPE_STATUS_DISCARDED)
     {
         mApp.getOverlayManager().recvFloodedMsg(msg, shared_from_this());
+
```

Comment:
same thing than for transactions: we need to do the accounting regardless of what the business logic decides about this SCP message. I imagine that we end up sharing the code.

For example: SCP messages get discarded for all sorts of reason; I'd rather keep track of "wasted" traffic (duplicate, but also propagated for no good reason).

Reply:
We also need to do this accounting for the `txSet` and `quorumSet` as we ask peers for it - and we need to know if we're getting one copy back or many copies (because we ask those things to too many peers for the same thing).

As I write this, I realize that maybe this logic can be simplified quite a bit and do something pretty close to "just tracking duplicates" (regardless of what the business logic does):
what we can do is do this type of accounting in `Peer::recvMessage(xdr::msg_ptr const& msg)` instead of relying on the floodmap: all we need to do is keep a small enough cache of `HASH(msg)` (LRU) and consider a message a duplicate if it's not in cache - we can use a shorthash here that should yield "good enough" results. If we want to only do this for some message types, all we need to do is make `Peer::recvMessage(StellarMessage const& stellarMsg)` return a bool (where returning true means "do the duplicate accounting").
We can then return true for the cases `TX_SET`, `TRANSACTION`, `SCP_QUORUMSET`, `SCP_MESSAGE`.